packages/reactstrap-validation-date/src/AvDateRange.js

```diff
@@ -0,0 +1,357 @@
+    };
+    if (this.props.distance) {
+      endValidate.distance = this.validateDistance;
+    }
+
+    const startId = `${(id || name).replace(/[^a-zA-Z0-9]/gi, '')}-start`;
+
+    const endId = `${(attributes.id || name).replace(
+      /[^a-zA-Z0-9]/gi,
+      ''
+    )}-end`;
+
+    const touched =
+      this.context.FormCtrl.isTouched(this.props.start.name) ||
+      this.context.FormCtrl.isTouched(this.props.end.name);
+    const hasError =
+      this.context.FormCtrl.hasError(this.props.start.name) ||
+      this.context.FormCtrl.hasError(this.props.end.name);
+    const isDirty =
+      this.context.FormCtrl.isDirty(this.props.start.name) ||
+      this.context.FormCtrl.isDirty(this.props.end.name);
+    const isBad =
+      this.context.FormCtrl.isBad(this.props.start.name) ||
+      this.context.FormCtrl.isBad(this.props.end.name);
+
+    const startDate = startValue ? moment(startValue, format) : null;
+    const endDate = endValue ? moment(endValue, format) : null;
+
+    const classes = classNames(
+      className,
+      touched ? 'is-touched' : 'is-untouched',
+      isDirty ? 'is-dirty' : 'is-pristine',
+      isBad ? 'is-bad-input' : null,
+      hasError ? 'av-invalid' : 'av-valid',
+      touched && hasError && 'is-invalid',
+      !startValue && !endValue && 'current-day-hightlight'
+    );
+
+    return (
+      <>
+        <AvInput
+          style={{ display: 'none' }}
+          {...this.props.start}
+          validate={{
+            date: true,
+            ...this.props.validate,
+            ...this.props.start.validate,
+          }}
+          value={this.state.startValue || ''}
+          type="text"
+          min={min}
+          max={max}
+          valueFormatter={this.valueFormatter}
+          valueParser={this.valueParser}
+        />
+        <AvInput
+          style={{ display: 'none' }}
+          {...this.props.end}
+          validate={{
+            date: true,
+            ...this.props.validate,
+            ...this.props.end.validate,
+          }}
+          value={this.state.endValue || ''}
+          min={min}
+          max={max}
+          valueFormatter={this.valueFormatter}
+          valueParser={this.valueParser}
+        />
+        <InputGroup
+          disabled={attributes.disabled}
+          className={classes}
+          onChange={({ target }) => {
+            if (target.id === startId || target.id === endId) {
+              this.onDatesChange(target.value);
+            }
+          }}
+          data-testid={`date-range-input-group-${name}`}
+        >
+          <DateRangePicker
+            disabled={attributes.disabled}
+            startDate={startDate}
+            startDateId={startId}
+            endDate={endDate}
+            endDateId={endId}
+            onDatesChange={this.onDatesChange}
+            focusedInput={focusedInput}
+            onFocusChange={this.onFocusChange}
+            isOutsideRange={isOutsideRange(min, max)}
```

Comment:
do we need the `min` and `max` props for this? or can we use the `distance` prop or `start` and `end` props to derive it? If we do need it, I think we should add to README. Same for `AvDate`

Reply:
Weird I think we never actually documented that you can input a `min` or `max` date in the component itself. Since that is something built into the `AvInput` field. 

We should probably do some logic to where we check if the `validate` prop has `max` and if not default to using the `max` from the standard props if that makes sense. I am seeing that if someone wants to limit the dates allowed in the datepicker but also want a custom error message they have to define it twice.